modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/constants/OAuth2ProviderRESTEndpointConstants.java

```diff
@@ -38,6 +38,9 @@
 	public static final String PROPERTY_KEY_CLIENT_REMOTE_HOST =
 		"client.remote.host";
 
+	public static final String PROPERTY_KEY_CLIENT_TRUSTED_APPLICATION =
+		"trusted_application";
+
```

Comment:
Here we are using the format of fields that can have a list of values (space delimited). See "token_introspection" which is a OAuth2Application "feature" for example. But we are storing it into a new model field.

I am wondering if you considered adding it as another "feature"? It would mean no breaking change to API which might be nice.

Reply:
Initially I was going to add it as a feature, it's something I talked to @csierra about, but thinking about it... it has no place as a feature since it really changes the OAuth2 flow, it makes sense that there's an API change (in this same API change I intend to add the other task, about "remember" per device)

Reply:
> @martamedio @stian-sigvartsen do you believe we need to split the pulls?

I don't see an easy way to split it, but maybe we could re-arrange some of the commits. But feel free to forward it as you prefer.

Reply:
so my take about this is that the token introspection is an OAuth2 feature that is optional and might be available or not but it is not related to the application definition in any way. In this case it is a characteristic of the client that alters how the flow works regarding one application.

I am weary that we turn that `feature` column into a new expando otherwise...

Reply:
what do you think @stian-sigvartsen?